packages/twenty-server/src/engine/core-modules/billing/services/billing-subscription.service.ts

```diff
@@ -164,42 +165,75 @@ export class BillingSubscriptionService {
         ? SubscriptionInterval.Month
         : SubscriptionInterval.Year;
 
-    const billingSubscriptionItem =
-      await this.getCurrentBillingSubscriptionItemOrThrow(workspace.id);
-
-    let productPrice;
+    const billingBaseProductSubscriptionItem =
+      await this.getCurrentBaseProductBillingSubscriptionItemOrThrow(
+        workspace.id,
+      );
 
     if (isBillingPlansEnabled) {
-      const baseProduct = await this.billingPlanService.getPlanBaseProduct(
-        BillingPlanKey.PRO,
+      const pricesPerPlan = await this.billingPlanService.getPricesPerPlan({
+        planKey: BillingPlanKey.PRO,
+        interval: newInterval,
+      });
+      const billingPricesPerPlanArray = [
+        pricesPerPlan.baseProductPrice,
+        ...pricesPerPlan.meteredProductsPrices,
+      ].filter((price): price is BillingPrice => price !== undefined);
```

Comment:
Could we filter with isDefined instead ? Seems like a better approach to narrow-down the type to BillingPrice[] instead of creating an anonymous typeguard that does the equivalent of isDefined.